cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/RunRecordMonitorService.java

```diff
@@ -188,7 +188,7 @@ public void removeRequest(ProgramRunId programRunId, boolean emitRunningChange)
     }
   }
 
-  private void emitMetrics(String metricName, long value) {
+  public void emitMetrics(String metricName, long value) {
```

Comment:
nit: I think it would be a little better to add a public `emitLaunchingMetric(long value)` method and keep this one private. That way the metric name stays a detail of this class instead of in both.

Reply:
Great suggestion. Done.